rita_client/src/operator_update/mod.rs

```diff
@@ -626,65 +596,39 @@ fn contains_forbidden_key(map: Map<String, Value>, forbidden_values: &[&str]) ->
     false
 }
 
-/// Given a vecdeque of usage hours, add up to a month's worth of hours to a returned vecdeque
-pub fn iterate_month_usage_data(mut data: VecDeque<RCUsageHour>) -> VecDeque<UsageHour> {
-    // one month in hours
-    let max_hour_iterations: u32 = 730;
-    let mut client_iter = 0;
-    let mut res = VecDeque::new();
-    while let Some(hour) = data.pop_front() {
-        // either we hit max iterations or we are on the second to last entry.
-        res.push_back(UsageHour {
-            up: hour.up,
-            down: hour.down,
-            price: hour.price,
-            index: hour.index,
-        });
-        if client_iter >= max_hour_iterations || data.len() == 1 {
-            break;
+/// This function handles preparing usage data for upload to operator tools
+fn prepare_usage_data_for_upload(
+    ops_last_seen_hour: Option<u64>,
+) -> Result<Option<UsageTrackerTransfer>, RitaClientError> {
+    // if this is none we will not send usage data to ops
+    // as it indicates we don't yet know what data ops has
+    match ops_last_seen_hour {
+        Some(ops_last_seen_hour) => {
+            let current_hour = get_current_hour()?;
+            let usage_data_client = get_usage_data_map(Client);
+            let usage_data_relay = get_usage_data_map(Relay);
+            // We check that the difference is >1 because we leave a 1 hour buffer to prevent from sending over an incomplete current hour
```

Comment:
Also I believe this comment is now stale as this is no longer checking for 1 hour behind (This may still be warranted- since part of the original problem was that routers were sending through incomplete hours, as the "current hour" is updated every couple seconds, and this can still happen if we send all the way up to the current hour. We're handling this on the ops side as well and only saving an hour behind the current to allow routers to complete their usage hours before we save the data but for the sake of not sending through hours that will not be saved may be worth changing)

Reply:
Unless we still want this data as displayed the way we have minute-spectrum with the latency graph, but I think that will need a different  struct as you mentioned way back

Reply:
I've modified the "prepare_usage_data_for_upload" function to avoid sending the current hour. Last 5 minutes data use is somthing we need to add to this though good catch